staging/src/k8s.io/client-go/pkg/api/types.go

```diff
@@ -1009,6 +1009,12 @@ type AzureFileVolumeSource struct {
 	// the ReadOnly setting in VolumeMounts.
 	// +optional
 	ReadOnly bool
+	// Directory permission code that gives rwx permissions to all users. Must be a value between 0 and 511(0777). Defaults to 511(0777).
```

Comment:
Valid range should be enforced in [validation.go](https://github.com/kubernetes/kubernetes/blob/448f8501596ef0c14e2720583385d5172f209105/pkg/api/validation/validation.go#L997) and default should be set in [defaults.go](https://github.com/kubernetes/kubernetes/blob/448f8501596ef0c14e2720583385d5172f209105/pkg/api/v1/defaults.go#L229)